### PR TITLE
[Enhancement] Support manual compaction sql interface for table & partition

### PR DESCRIPTION
**Pull request by meegoo**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

```
mysql> CREATE TABLE ss( event_day DATE, pv BIGINT) DUPLICATE KEY(event_day) PARTITION BY date_trunc('month', event_day) DISTRIBUTED BY HASH(event_day) BUCKETS 8 PROPERTIES("replication_num" = "3");
Query OK, 0 rows affected (0.04 sec)

mysql> insert into ss values('2023-02-14', 2),('2033-03-01',2);
Query OK, 2 rows affected (0.08 sec)
{'label':'insert_734648fa-c878-11ed-90d6-00163e0dcbfc', 'status':'VISIBLE', 'txnId':'5008'}

mysql> insert into ss values('2023-02-14', 2),('2033-03-01',2);
Query OK, 2 rows affected (0.05 sec)
{'label':'insert_85c95c1b-c878-11ed-90d6-00163e0dcbfc', 'status':'VISIBLE', 'txnId':'5009'}

mysql> alter table ss compact p202303;
ERROR 1064 (HY000): Unexpected exception: Partition not found: p202303

mysql> alter table ss compact p203303;
Query OK, 0 rows affected (0.00 sec)

mysql> alter table ss compact;
Query OK, 0 rows affected (0.01 sec)

mysql> alter table ss compact (p202302,p203303);
Query OK, 0 rows affected (0.00 sec)

mysql> alter table ss cumulative compact (p202302,p203303);
Query OK, 0 rows affected (0.00 sec)

mysql> alter table ss base compact (p202302,p203303);
Query OK, 0 rows affected (0.00 sec)
```
## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by meegoo** [comment]
> how about using `alter table ss compact p1,p2,p3`?

It's ok, but currently all syntax for operating multiple partitions uses `()`
So stay consistent here

**Comment by LiShuMing** [comment]
`alter table ss compact p203303;`
is it blocked until the compaction is done? or async compaction?
can we know if it has done the compaction by using SQL?

**Comment by meegoo** [comment]
> `alter table ss compact p203303;` is it blocked until the compaction is done? or async compaction? can we know if it has done the compaction by using SQL?

async compaction, user can query related information from be_tablets/be_compactions in information_schema

File: be/src/agent/agent_task.cpp [hunk 1]
```diff
@@ -455,6 +456,31 @@ void run_check_consistency_task(const std::shared_ptr<CheckConsistencyTaskReques
     remove_task_info(agent_task_req->task_type, agent_task_req->signature);
 }
 
+void run_compaction_task(const std::shared_ptr<CompactionTaskRequest>& agent_task_req, ExecEnv* exec_env) {
```
Comment (chaoyli) [inline be/src/agent/agent_task.cpp:459]: What's the difference between this pull request and #21094?

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 2]
```diff
@@ -790,6 +791,10 @@ rollupRenameClause
     : RENAME ROLLUP rollupName=identifier newRollupName=identifier
     ;
 
+compactionClause
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:794]: I think `compact` is enough. The user should not take base/cumulative into consideration.

Comment (meegoo) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:794]: base/cumulative is optional

File: be/src/agent/agent_task.cpp (hunk 1 above)
Comment (meegoo) [reply be/src/agent/agent_task.cpp:459]: This granularity is table and partition

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 (hunk 2 above)
Comment (chaoyli) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:810]: I think we can auto-determine to do cumulative compaction or base compaction.

File: be/src/agent/agent_task.cpp (hunk 1 above)
Comment (chaoyli) [reply be/src/agent/agent_task.cpp:459]: `alter table ss compact partition p203303;`
How about add a partition keyword?

**Review by gengjun-git — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by jaogoy** [comment]
@meegoo will this PR fix #17270?
and, where is the docs (including the syntax)?

**Pull request merged**